README.Docker.md

```diff
@@ -0,0 +1,246 @@
+remove the socket, the path for the socket is to be volume-mounted, to expose
+it for others on the same host to use. The following volume can be used for
+this purpose. Do ensure that the directory on the host actually exists and
+clamav inside the container has permission to access it.
+Caution is required when managing permissions, as incorrect permission could
+open clamd for anyone on the host system.
+```console
+       --volume '/var/lib/docker/data/clamav/sockets/:/run/clamav/'
+```
+
+With the socket exposed to the host, any other service can now talk to `clamd`
+as well. If for example `clamdtop` where installed on the local host, calling
+```console
+clamdtop "/var/lib/docker/data/clamav/sockets/clamd.sock"
+```
+should work just fine. Likewise, running `clamdtop` in a different container,
+but sharing the socket will equally work. While `clamdtop` works well as an
+example here, it is of course important to realize, this can also be used to
+connect a mail server to `clamd`.
+
+
+### TCP/UDP
+While `clamd` and `milterd` will listen on the default tcp ports as per
+configuration directives, docker does not expose these by default to the host.
+Only within  containers can these ports be accessed. To expose, or publish,
+these ports to the host, and thus potentially over the (inter)network, the
+`--publish` (or `--publish-all`) flag to `docker run` can be used.
+While more advanced/secure mappings can be done as per documentation, the basic
+way is to `--publish [<host_port>:]<container_port>` to make the port available
+to the host.
+```console
+       --publish 73310:3310 \
+       --publish 7357
+```
+The above would thus publish the milter port 3310 as 73310 on the host and the
+clamd port 7357 as a random to the host. The random port can be inspected via
+`docker ps`.
+
+**Warning** extreme caution is to be taken when using `clamd` over TCP/UDP as
+there are no protections on that level. All traffic is un-encrypted. Extra care
+is to be taken when using TCP/UDP communications.
+
+
+## Virus database
+The virus database in `/usr/share/clamav` is by default unique to each container
+and thus is normally not shared. For simple setups this is fine, where only one
+instance of `clamd` is expected to run in a dockerized environment. However
+some use cases may want to efficiently share the database. To do so, the
+`docker volume` command can be used to create a persistent virus database.
+
+
+## Container clamd healthcheck
+Docker has the ability to run simple `ping` checks on services running inside
+containers. If clamd is running inside the container, docker will occasionally
+send a `ping` to clamd on the default port and wait for the pong from clamd.
+If clamd fails to respond, docker will treat this as an error. The healthcheck
+results can be viewed with `docker inspect`.
+
+
+## Performance
+The performance impact of running `clamd` in docker is negligible. Docker is
```

Comment:
Love it! Thanks.

Overall this documentation is amazing.  Thank you so much for the great work, and apologies it's taking me so long to review each time.

Reply:
haha, hopefully round 3 (or 4?) we can count in days rather then months :p

Reply:
Actually, it took far less time then I thought (had a little window, and was quite quick :p)